frontend/views/components/GroupSettings/InviteLinks.vue

```diff
@@ -109,72 +114,94 @@ export default {
-            inviteLink: 'http://localhost:8000/app/join?groupId=21XWnNFz7RbNPKHUqAeSLLT1cNHnnCssmSw6dJeB1gfSSeZc7v&secret=4460',
-            isAnyoneLink: false,
-            state: {
-              description: 'Not used yet',
-              expireInfo: '1d 2h 30m left',
-              isExpired: false
-            }
-          },
-          {
-            name: 'Brian Eno',
-            inviteLink: 'http://localhost:8000/app/join?groupId=30aFnTYz7RbqAeSLLT1cNfSSeZHN6KHUnnCssmSw6dJeB1gc7v&secret=2250',
-            isAnyoneLink: false,
-            state: {
-              description: 'Used',
-              expireInfo: '',
-              isExpired: true
-            }
-          },
-          {
-            name: 'Carl Sagan',
-            inviteLink: 'http://localhost:8000/app/join?groupId=B1gfSSeZc721XWnNFz7RbkoyHUqAeSwLT1cNHnnCssmSw6dJev&secret=1348',
-            isAnyoneLink: false,
-            state: {
-              description: 'Not used',
-              expireInfo: 'Expired',
-              isExpired: true
-            }
-          },
-          {
-            name: 'Anyone',
-            inviteLink: 'http://localhost:8000/app/join?groupId=s8LT1cNHnnCs21XWnNFz7RbNPKHUqAesmSw6dJeB1gfSSeZc7v&secret=5521',
-            isAnyoneLink: true,
-            state: {
-              description: '10/60 used',
-              expireInfo: 'Expired',
-              isExpired: true
-            }
-          }
-        ]
+        }
       }
     }
   },
   methods: {
     unfocusSelect () {
       this.$refs.select.blur()
     },
-    toInvite (e) {
-      if (e.target.tagName === 'SPAN') this.$router.push({ path: '/invite' })
-    },
     activateWebShare (inviteLink) {
       if (navigator.share) {
         navigator.share({
-          title: this.L('Your invite'),
+          title: L('Your invite'),
           url: inviteLink
         })
       }
+    },
+    inviteStatusDescription ({
+      isAnyoneLink,
+      isInviteExpired,
+      isAllInviteUsed,
+      quantity,
+      numberOfResponses
+    }) {
+      if (isAnyoneLink) return L('{numberOfResponses}/{quantity} used', { numberOfResponses, quantity })
+      else if (isAllInviteUsed) return L('Used')
+      else return isInviteExpired ? L('Not used') : L('Not used yet')
+    },
+    readableExpiryInfo (expiryTime) {
+      const MIL = 1000
+      const MIL_MIN = 60 * MIL
+      const MIL_HR = MIL_MIN * 60
+      const MIL_DAY = 24 * MIL_HR
+      let remainder
+
+      const days = Math.floor(expiryTime / MIL_DAY)
+      remainder = expiryTime % MIL_DAY
+      const hours = Math.floor(remainder / MIL_HR)
+      remainder = remainder % MIL_HR
+      const minutes = Math.ceil(remainder / MIL_MIN)
+      const options = [
+        undefined,
+        L('{minutes}m left', { minutes }),
+        L('{hours}h {minutes}m left', { hours, minutes }),
+        L('{days}d {hours}h {minutes}m left', { days, hours, minutes })
+      ]
+
+      return options[!!days + !!hours + !!minutes]
```

Comment:
@SebinSong this line is pure genius, did you come up with this yourself?

Reply:
It was inspired by @sandrina-p 's comment [here](https://github.com/okTurtles/group-income-simple/pull/742#discussion_r340702726)

Reply:
@taoeffect was talking about the fact you sum boolean to access an array. That's not only genius, it's also madness x)

Reply:
@sandrina-p Thanks again for the inspiration! 👍 (I keep saying this but I'm learning a lot from the project)

Reply:
@SebinSong Alas, I think there is still some math issue here, because if there is 1 day, 0 hours, and 30 minutes left, it will display `0h 30m left` (which is incorrect) instead of either `1d 0h 30m left` or `1d 30m left`

Reply:
You are right. Fixed! :)